docs/ucsmsdk_ug.rst

```diff
@@ -0,0 +1,913 @@
+    cd ucsmsdk 
+    sudo make install
+
+Uninstallation
+--------------
+
+Uninstallation Steps
+~~~~~~~~~~~~~~~~~~~~
+
+Irrespective of the method that was used to install the SDK, it can be
+uninstalled using the below command,
+
+::
+
+    pip uninstall ucsmsdk
+
+Getting Started
+---------------
+
+Connecting Disconnecting
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    from ucsmsdk.ucshandle import UcsHandle
+
+    # Create a connection handle
+    handle = UcsHandle("192.168.1.1", "admin", "password")
+
+    # Login to the server
+    handle.login()
+
+    # Logout from the server
+    handle.logout()
+
+Refer `UcsHandle API
+Reference <https://ciscoucs.github.io/ucsmsdk_docs/ucsmsdk.html#module-ucsmsdk.ucshandle>`__
+for detailed parameter sets to ``UcsHandle``
+
+Base APIs
+~~~~~~~~~
+
+The SDK provides APIs to enable CRUD operations.
+
+-  **C**\ reate an object - ``add_mo``
+-  **R**\ etrieve an object -
+   ``query_dn``,\ ``query_classid``,\ ``query_dns``,\ ``query_classids``
+-  **U**\ pdate an object - ``set_mo``
+-  **D**\ elete an object - ``delete_mo``
+
+The above APIs can be bunched together in a transaction (All or None).
+``commit_mo`` commits the changes made using the above APIs.
+
+All these methods are invoked on a ``UcsHandle`` instance. We refer it
+by ``handle`` in all the examples here-after. Refer to the `Connecting
+Disconnecting <#connecting-disconnecting>`__ to create a new handle.
+
+Creating Objects
+~~~~~~~~~~~~~~~~
+
+Creating managed objects is done via ``add_mo`` API.
+
+Example:
+
+The below example creates a new Service Profile(\ ``LsServer``) Object
+under the parent ``org-root``
+
+::
+
+    from ucsmsdk.mometa.ls.LsServer import LsServer
+
+    sp = LsServer(parent_mo_or_dn="org-root", name="sp_demo")
+    handle.add_mo(sp)
+
+**note**: the changes will only be sent to server when
+``handle.commit()`` is called.
+
+`Add Mo API
+reference <https://ciscoucs.github.io/ucsmsdk_docs/ucsmsdk.html?highlight=add_mo#ucsmsdk.ucshandle.UcsHandle.add_mo>`__
+
+Querying Objects
+~~~~~~~~~~~~~~~~
+
+-  Querying Objects via Distinguished Name (DN)
+
+   ::
+
+       object = handle.query_dn("org-root/ls-sp_demo")
+
+-  Querying Multiple Objects via Multiple Distinguished Names (DN)
+
+   The returned object is a dictionary in ``{dn:object}`` format
+
+   ::
+
+       object_dict = handle.query_dn("org-root/ls-sp_demo1", "org-root")
+
+-  Querying Objects via class Id
+
+   The below returns all objects of type ``orgOrg``
+
+   ::
+
+       object_array = handle.query_classid("orgOrg")
+
+-  Querying Objects via multiple class Ids
+
+   The below returns all objects of type ``orgOrg`` and ``fabricVlan``.
+   The output is a dictionary of format ``{classId: [objects]}``
+
+   ::
+
+       object_dict = handle.query_classid("orgOrg", "fabricVlan")
+
+`Query DN API
+reference <https://ciscoucs.github.io/ucsmsdk_docs/ucsmsdk.html?highlight=add_mo#ucsmsdk.ucshandle.UcsHandle.query_dn>`__
```

Comment:
Here the highlighted keyword is wrong.
In the URL we have given "highlight=add_mo" but instead it should be "query_dn".

Similar issues are done at couple of more locations in the document.